subsystem/src/manager.rs

```diff
@@ -226,15 +226,31 @@ impl Manager {
         self.add_subsystem_with_custom_eventloop(
             "ctrl-c",
             |mut call_rq: CallRequest<()>, mut shutdown_rq| async move {
+                // Gracefully handle SIGTERM on *nix
+                #[cfg(unix)]
+                let term_signal = async move {
+                    let mut signal =
+                        tokio::signal::unix::signal(tokio::signal::unix::SignalKind::terminate())
+                            .expect("Signal handler failed unexpectedly");
+                    signal.recv().await;
+                };
```

Comment:
I think the `async` scope and the final `await` cancel out, don't they?

Reply:
Right, but this does not work:
```
let term_signal = {
    let mut signal =
        tokio::signal::unix::signal(tokio::signal::unix::SignalKind::terminate())
            .expect("Signal handler failed unexpectedly");
    signal.recv()
};
```

The error:
```
`signal` does not live long enough
borrowed value does not live long enough
```

Reply:
And I can't move `tokio::signal::unix` outside of `#[cfg(unix)]`-only block.

Reply:
ok, thanks for explanation